core/src/main/java/info/nightscout/androidaps/utils/ui/TargetBgProfileGraph.kt

```diff
@@ -0,0 +1,99 @@
+package info.nightscout.androidaps.utils.ui
+
+import android.content.Context
+import android.util.AttributeSet
+import com.jjoe64.graphview.DefaultLabelFormatter
+import com.jjoe64.graphview.GraphView
+import info.nightscout.androidaps.core.R
+import info.nightscout.androidaps.interfaces.GlucoseUnit
+import info.nightscout.androidaps.interfaces.Profile
+import info.nightscout.androidaps.utils.Round
+import info.nightscout.androidaps.plugins.general.overview.graphExtensions.AreaGraphSeries
+import info.nightscout.androidaps.plugins.general.overview.graphExtensions.DoubleDataPoint
+import java.text.NumberFormat
+import java.util.ArrayList
+import kotlin.math.max
+
+class TargetBgProfileGraph : GraphView {
+
+    constructor(context: Context?) : super(context)
+    constructor(context: Context?, attrs: AttributeSet?) : super(context, attrs)
+
+    fun show(profile: Profile) {
+
+        removeAllSeries()
+        val targetArray: MutableList<DoubleDataPoint> = ArrayList()
+        var maxValue = 0.0
+        val units = profile.units
+        for (hour in 0..23) {
+            val valueLow = Profile.fromMgdlToUnits(profile.getTargetLowMgdlTimeFromMidnight(hour * 60 * 60), units)
+            val valueHigh = Profile.fromMgdlToUnits(profile.getTargetHighMgdlTimeFromMidnight(hour * 60 * 60), units)
+            maxValue = max(maxValue, valueHigh)
+            targetArray.add(DoubleDataPoint(hour.toDouble(), valueLow, valueHigh))
+            targetArray.add(DoubleDataPoint((hour + 1).toDouble(), valueLow, valueHigh))
+        }
+        val targetDataPoints: Array<DoubleDataPoint> = Array(targetArray.size) { i -> targetArray[i] }
+        val targetSeries: AreaGraphSeries<DoubleDataPoint> = AreaGraphSeries(targetDataPoints)
```

Comment:
ISF has backround, IC has not. Voting for graph without

Reply:
ISF should have not background, basal and target range area should

Reply:
fixed